src/components/input/inputs.tsx

```diff
@@ -0,0 +1,121 @@
+import { Component, h, Prop, Element, State } from "@stencil/core";
+
+export type InputType = 'text' | 'textarea';
+
+@Component({
+  tag: 'bds-input',
+  styleUrl: 'input.scss',
+  shadow: true
+})
+export class Input {
+  @Element() element: HTMLElement;
+
+  @Prop() inputId!: string;
+  @Prop() inputName?: string = '';
+
+  @Prop() type?: InputType = 'text';
+  @Prop() label?: string = '';
+  @Prop() placeholder?: string = '';
+  @Prop() helperMessage?: string = '';
+  @Prop() errorMessage?: string = '';
+
+  @Prop() iconLeft?: string = '';
+  @Prop() iconRight?: string = '';
+
+  @Prop({ reflect: true }) value?: string = '';
+
+  @Prop({ reflect: true }) danger?: boolean = false;
+
+  @Prop() onChangeValue: Function;
+
+  @State() isPressed?= false;
+
+  renderIconLeft(): HTMLElement {
+    return this.iconLeft && (
+      <div class={{
+        "input__icon-left": true,
+        "input__icon-left--large": !!this.label,
+      }}>
+        <bds-icon
+          size={this.label ? "medium" : "small"}
+          name={this.iconLeft}
+          color="inherit">
+        </bds-icon>
+      </div>
+    )
+  }
+
+  renderLabel(): HTMLElement {
+    return this.label && (
+      <label class="input__container__label">
+        <bds-typo variant="fs-12" bold="bold">{this.label}</bds-typo>
+      </label>
+    )
+  }
+
+  renderMessageError(): HTMLElement {
+    if (!this.danger && this.helperMessage) {
+      return (
+        <div class="input__message">
+          <bds-typo variant="fs-12">{this.helperMessage}</bds-typo>
+        </div>
+      )
+    }
+
+    if (this.danger && this.errorMessage) {
+      return (
+        <div class="input__message input__message--danger">
+          <div class="input__message__icon">
+            <bds-icon
+              size="x-small"
+              name="error"
+              theme="solid"
+              color="inherit">
+            </bds-icon>
+          </div>
+          <bds-typo variant="fs-12">{this.errorMessage}</bds-typo>
+        </div>
+      )
+    }
+
+    return null;
```

Comment:
We can adopt convention to use only `null` or `undefined`, not both. In our context we will not need to differentiate them in the most cases. We already use it at `BlipPortal` and you can add this convention in eslint file.

Reply:
And I suggest `undefined` 😃